base/logger.go

```diff
@@ -34,3 +38,45 @@ func logCollationWorker(collateBuffer chan string, logger *log.Logger, bufferSiz
 		}
 	}
 }
+
+func runLogDeletion(logDirectory string, logLevel string, maxSize int) {
+
+	maxSize = maxSize * 1048576 //Convert MB input to bytes
+
+	logDirectory = logDirectory + "/"
+
+	logDiff := 0 - maxSize
+
+	files, err := ioutil.ReadDir(logDirectory)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	for _, f := range files {
+		var extension = filepath.Ext(logDirectory + f.Name())
+		if extension == ".gz" && strings.Contains(f.Name(), logLevel) {
+			logDiff += int(f.Size())
+		}
+	}
+
+	var modTime time.Time
+	var oldestFile string
+	var fileSize int
+	for logDiff > 0 {
+		files, _ = ioutil.ReadDir(logDirectory)
+		for _, f := range files {
```

Comment:
Not sure we can? We need to know how much we are over the limit before continuing to find the oldest files and delete. I think it may be possible to do it in one explicit loop but it would require saving an ordered list of files which would end up being a loop to sort anyways?